src/lua/init_path.rs

```diff
@@ -0,0 +1,51 @@
+//! Contains methods for initializing the lua config
+
+use std::fs::{self, OpenOptions, File};
+use std::env;
+use std::path::Path;
+
+use std::io::Result as IOResult;
+
+pub const INIT_FILE: &'static str = "init.lua";
+pub const INIT_FILE_FALLBACK_PATH: &'static str = "/etc/way-cooler/";
+
+#[inline]
+fn read_file<P: AsRef<Path>>(path: P) -> IOResult<File> {
+    OpenOptions::new().read(true).open(path)
+}
+
+/// Parses environment variables
+pub fn get_config() -> Result<File, &'static str> {
+    if let Ok(path_env) = env::var("WAY_COOLER_CONFIG") {
```

Comment:
It would perhaps be cleaner to implement this with a for loop iterating over a tuple or hard-coded array of paths. It would *maybe* be slightly less efficient by reading those variables and paths at the start, but it's doubtful to make any kind of difference. And you can improve that by making it iterate over a lazy iterator and it'd probably compile down to similarly efficient code.